### PR TITLE
Fix #443: profile composition

### LINKED ISSUE #443: composition of test profiles
Currently we support custom profiles, e.g. we use different profiles for CI, open PRs and merged PRs.

If we look at `profiles/default.yml` and `profiles/pull.yml`, we would find many duplications. A solution to avoid the duplication is to support composition of profiles.

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #443: profile composition

### CODE REVIEW

File: bin/master [hunk 1]
```diff
@@ -70,7 +70,14 @@ do
     # Achieve the above by
     # 1. remember the last scheduled commit
     # 2. schedule new jobs from the last scheduled commit
-    echo "bin/gauge $pr $commit && bin/deploy" > "$PROG_HOME/jobs/work-merge-$pr.job"
+    job="$PROG_HOME/jobs/work-merge-$pr.job"
+    test_sbt=$((RANDOM % 100 < 10))
```
Comment (nicolasstucki) [inline bin/master:74]: What is this for? Maybe add come comment.

File: .gitignore [hunk 2]
```diff
@@ -82,3 +82,5 @@ logs/*.out
 
 .sbt/
 .ivy2/
+*.plan
+*.js
```
Comment (nicolasstucki) [inline .gitignore:86]: Why did you add `.js` if we do have some `js` files?

**Review by nicolasstucki — COMMENTED** [review]
Otherwise LGTM

File: .gitignore (hunk 2 above)
Comment (liufengyun) [reply .gitignore:86]: The reason is that the command `bin/sync-profiles` will generate `.js` for `ci.yml`, which is never needed.

File: bin/master (hunk 1 above)
Comment (liufengyun) [reply bin/master:74]: This is to only test `dotty-sbt` every 10 PRs. I'll add a comment.

Comment (nicolasstucki) [reply bin/master]: Ok

**Review by nicolasstucki — APPROVED** [review]
